apps/worker/src/app/workflow/usecases/execute-bridge-job/execute-bridge-job.usecase.ts

```diff
@@ -137,8 +137,13 @@ export class ExecuteBridgeJob {
   private async findControlVariables(command: ExecuteBridgeJobCommand, workflow: NotificationTemplateEntity) {
     const controls = await this.controlVariablesRepository.findOne({
       _organizationId: command.organizationId,
-      _workflowId: workflow._id,
-      _stepId: command.job.step._id,
+      _environmentId: command.environmentId,
+      /*
+       * TODO: something odd was happening with stepId incorrectly matching.
+       * Investigate why this is, it causes invalid state hydration errors in Framework.
+       */
```

Comment:
This should be fixed separately. My assumption is that multiple syncs in the same environment for Workflows with the same identifier cause a corrupt controls state, where writing controls does not override the MongoDB `stepId` used by old steps.

Reply:
This was resolved in a separate PR.